compiler/haskell-ide-core/src/Development/IDE/State/Shake.hs

```diff
@@ -336,9 +337,8 @@ defineEarlyCutoff op = addBuiltinRule noLint noIdentity $ \(Q (key, file)) old m
             (bs, res) <- actionCatch
                 (do v <- op key file; liftIO $ evaluate $ force v) $
                 \(e :: SomeException) -> pure (Nothing, ([ideErrorText file $ T.pack $ show e | not $ isBadDependency e],Nothing))
-            res <- return $ first (map $ fixDiagnostic file) res
+            res@((badErrors,_),_) <- return $ first (addDiagnostics file) res
```

Comment:
How are these errors "bad" in the sense listed below? We now aren't catching errors which have noRange?

Reply:
They're diagnostics which originate in the package not being run by shake, which is not what we expect

Reply:
From reading the definition of addDiagnostics I understand, and dislike :)